src/main/scala/cromwell/engine/db/slick/DataAccessController.scala

```diff
@@ -0,0 +1,310 @@
+          row.wdlUri === id
+        }
+        reps.reduceLeftOption(_ || _).getOrElse(true: Rep[Boolean])
+      }
+    }
+
+    if (states.isDefined) {
+      query = query.filter { row =>
+        val reps = states.get.map { id =>
+          row.status === id.toString
+        }
+        reps.reduceLeftOption(_ || _).getOrElse(true: Rep[Boolean])
+      }
+    }
+
+    if (beforeStart.isDefined) {
+      query = query.filter(_.startDt < date2timestamp(beforeStart.get))
+    }
+
+    if (afterStart.isDefined) {
+      query = query.filter(_.startDt > date2timestamp(afterStart.get))
+    }
+
+    if (beforeEnd.isDefined) {
+      query = query.filter(_.endDt < date2timestamp(beforeEnd.get))
+    }
+
+    if (afterEnd.isDefined) {
+      query = query.filter(_.endDt > date2timestamp(afterEnd.get))
+    }
+
+    // TODO: For now, running queries in separate transactions, and use futures to compose the results.
+    // Find more elegant one-to-many query option.
+    // In 2.1 we could pass in a transaction session. Not sure how to do this in 3.0.
+    // It may be that Slick's composition may just "work" this way.
+    // http://slick.typesafe.com/doc/3.0.0/orm-to-slick.html#navigating-the-object-graph
+    // http://stackoverflow.com/questions/30567365/slick-3-0-0-how-to-query-one-to-many-many-to-many-relations
+
+    val workflowExecutionRowsFuture = database.run(query.result.transactionally)
+    val resultsFuture = workflowExecutionRowsFuture flatMap { workflowExecutionRows =>
+
+      val results = workflowExecutionRows map { workflowExecutionRow =>
+
+        // Query all the symbol store entries
+        val symbolsQuery = dataAccess.symbols.filter(
+          _.workflowExecutionId === workflowExecutionRow.workflowExecutionId.get)
+        val symbolsFuture: Future[Seq[Symbol]] = database.run(symbolsQuery.result.transactionally)
+
+        // Query all the local jobs
+        val localJobQuery = for {
+          executionRow <- dataAccess.executions.filter(
+            _.workflowExecutionId === workflowExecutionRow.workflowExecutionId.get)
+          localJobRow <- dataAccess.localJobs.filter(
+            _.executionId === executionRow.executionId)
+        } yield (executionRow, localJobRow)
+        val localJobQueryFuture = database.run(localJobQuery.result.transactionally)
+
+        // Query all the jes jobs
+        val jesJobQuery = for {
+          executionRow <- dataAccess.executions.filter(
+            _.workflowExecutionId === workflowExecutionRow.workflowExecutionId.get)
+          jesJobRow <- dataAccess.jesJobs.filter(
+            _.executionId === executionRow.executionId)
+        } yield (executionRow, jesJobRow)
+        val jesJobQueryFuture = database.run(jesJobQuery.result.transactionally)
+
+        val rowTuples = for {
+          symbols <- symbolsFuture
+          jesJobs <- jesJobQueryFuture
+          localJobs <- localJobQueryFuture
+        } yield (symbols, jesJobs, localJobs)
```

Comment:
If @kshakir hasn't done this already I can, as I'm adding the aux table row here.